src/display_list/from_snippet.rs

```diff
@@ -2,6 +2,52 @@
 use super::*;
 use crate::{formatter::get_term_style, snippet};
 
+struct CursorLines<'a>(&'a str);
+
+impl<'a> CursorLines<'a> {
+    fn new(src: &str) -> CursorLines<'_> {
+        CursorLines(src)
+    }
+}
+
+#[repr(u8)]
+enum EndLine {
+    EoF = 0,
+    One = 1,
+    Two = 2,
+}
```

Comment:
Why do you use `repr(u8)` on this?
Also, can you name it CRLF and LF respectively? EOF is also all capitals.

Reply:
It's done